packages/components/src/toggle-group-control/toggle-group-control/utils.ts

```diff
@@ -21,16 +21,21 @@ type ValueProp = ToggleGroupControlProps[ 'value' ];
 export function useComputeControlledOrUncontrolledValue(
 	valueProp: ValueProp
 ): { value: ValueProp; defaultValue: ValueProp } {
+	const isInitialRender = useRef( true );
 	const prevValueProp = usePrevious( valueProp );
 	const prevIsControlled = useRef( false );
 
+	useEffect( () => {
+		if ( isInitialRender.current ) {
+			isInitialRender.current = false;
+		}
+	}, [] );
+
 	// Assume the component is being used in controlled mode on the first re-render
 	// that has a different `valueProp` from the previous render.
 	const isControlled =
 		prevIsControlled.current ||
-		( prevValueProp !== undefined &&
-			valueProp !== undefined &&
-			prevValueProp !== valueProp );
+		( ! isInitialRender.current && prevValueProp !== valueProp );
```

Comment:
In the initial render, it is still unknown whether we're in controlled or uncontrolled mode. For all intents and purposes, they are the same because we just consider the given `value` prop as the initial value. We just don't want to say that we're `isControlled` yet, since this boolean is effectively irreversible once we set it to true.

Also the `prevValueProp !== valueProp` check is meaningless in the initial render because the previous value will always be undefined.

Reply:
Makes sense.